extension/src/cli/viewable.ts

```diff
@@ -0,0 +1,92 @@
+import { Event, EventEmitter } from 'vscode'
+import { CliEvent, CliResult, CliStarted } from '.'
+import {
+  captureStdErr,
+  notifyCompleted,
+  notifyOutput,
+  notifyStarted
+} from './util'
+import { getCommandString } from './command'
+import { ProcessOptions, createProcess } from '../process/execution'
+import { StopWatch } from '../util/time'
+import { PseudoTerminal } from '../vscode/pseudoTerminal'
+import { DeferredDisposable } from '../class/deferred'
+
+export class ViewableCliProcess extends DeferredDisposable {
+  public readonly onDidDispose: Event<void>
+
+  private readonly pseudoTerminal: PseudoTerminal
+
+  constructor(
+    termName: string,
+    options: ProcessOptions,
+    processStarted: EventEmitter<CliStarted>,
+    processCompleted: EventEmitter<CliResult>
+  ) {
+    super()
+    const processOutput = this.dispose.track(new EventEmitter<string>())
+    const terminalClosed = this.dispose.track(new EventEmitter<void>())
+    const onDidCloseTerminal = terminalClosed.event
+
+    this.pseudoTerminal = this.dispose.track(
+      new PseudoTerminal(processOutput, terminalClosed, termName)
+    )
+
+    this.pseudoTerminal.setBlocked(true)
+    void this.show().then(() => this.deferred.resolve())
+
+    this.createProcess(options, processStarted, processOutput, processCompleted)
+
+    const disposed = this.dispose.track(new EventEmitter<void>())
+    this.onDidDispose = disposed.event
+
+    this.dispose.track(
+      onDidCloseTerminal(() => {
+        disposed.fire()
+        this.dispose()
+      })
+    )
+  }
+
+  public show() {
+    return this.pseudoTerminal.openCurrentInstance()
+  }
+
+  private createProcess(
+    options: ProcessOptions,
+    processStarted: EventEmitter<CliEvent>,
+    processOutput: EventEmitter<string>,
+    processCompleted: EventEmitter<CliResult>
+  ) {
+    processOutput.fire(
+      `Running: ${
+        options.executable === 'git' ? 'git' : 'dvc'
+      } ${options.args.join(' ')}\r\n\n`
```

Comment:
[I] When we call a dvc as python module this is going to be cooked just send the command.